### PR TITLE
fix: add access client in parse clients

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue
Closes: #1023

## Approach
Add a function to access client in parse clients.

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Review by chadpav — APPROVED** [review]
Seems pretty straighforward.

**Comment by mtrezza** [comment]
Could you rephrase the PR title to better understand the new feature? Maybe also add the class or method name.

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [8.0.1](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-8.0.0...flutter-8.0.1) (2024-11-28)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]: If this is a feature PR, then the version should be 8.1.0

File: packages/dart/lib/src/network/parse_dio_client.dart [hunk 2]
```diff
@@ -14,6 +14,8 @@ class ParseDioClient extends ParseClient {
 
   late _ParseDioClient _client;
 
+  dio.Dio get dioClient => _client;
```
Comment (mtrezza) [inline packages/dart/lib/src/network/parse_dio_client.dart:17]: Code cov missing, add tests?

File: packages/dart/CHANGELOG.md (hunk 1 above)
Comment (mbfakourii) [reply packages/dart/CHANGELOG.md]: added.

File: packages/dart/lib/src/network/parse_dio_client.dart (hunk 2 above)
Comment (mbfakourii) [reply packages/dart/lib/src/network/parse_dio_client.dart:17]: added.

File: packages/dart/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-## [8.0.1](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-8.0.0...flutter-8.0.1) (2024-11-28)
+## [8.1.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-8.0.0...flutter-8.1.0) (2024-11-28)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [8.1.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-8.0.0...flutter-8.1.0) (2024-12-18)
```

File: packages/dart/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+## [8.1.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-8.0.0...flutter-8.1.0) (2024-12-18)
+
+### Features
+
+* Add access client in parse clients ([#1025](https://github.com/parse-community/Parse-SDK-Flutter/pull/1025))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Add client access via `ParseDioClient.dioClient` and `ParseHTTPClient.httpClient` ([#1025](https://github.com/parse-community/Parse-SDK-Flutter/pull/1025))
```

**Review by mtrezza — APPROVED** [review]
Looks good; if the new PR title makes sense, then I'll go ahead and merge.

**Review by mtrezza — COMMENTED** [review]
Just 1 thing:

Would it make sense to name the properties just `client` as in `ParseDioClient.client`, `ParseHTTPClient.client`? The class names already say what type of client it is.

**Comment by mbfakourii** [comment]
> Just 1 thing:
> 
> Would it make sense to name the properties just `client` as in `ParseDioClient.client`, `ParseHTTPClient.client`? The class names already say what type of client it is.

Good suggestion.

**Review by mtrezza — APPROVED** [review]
Looks good!
